File: sseni/2024-06/P250137_붕대감기.java
package com.example.algorithm.study;

public class P250137_붕대감기 {
    public static int solution(int[] bandage, int health, int[][] attacks) {
        int now = health - attacks[0][1];

        if (now <= 0) return -1;

        for (int i = 1; i < attacks.length; i++) {
            int tmp = attacks[i][0] - attacks[i - 1][0] - 1;
            int recovery = ((int) (tmp / bandage[0])) * bandage[2];

            now += bandage[1] * tmp + recovery;

            if (now > health) now = health;

            now -= attacks[i][1];

            if (now <= 0) return -1;
        }

        return now;
    }

    public static void main(String[] args) {
        System.out.println(solution(new int[]{5, 1, 5}, 30, new int[][]{{2, 10}, {9, 15}, {10, 5}, {11, 5}}));
    }
}

File: seulgi/2024-06/P250137_붕대감기.py
def solution(bandage, health, attacks):
    max_health = health
    time = 0
    conti_time = 0

    for i in range(len(attacks)):
        while True:
            time += 1
            conti_time += 1

            if attacks[i][0] == time:
                health -= attacks[i][1]
                conti_time = 0
                break
            else:
                if health != max_health:
                    if conti_time == bandage[0]:
                        health = health+ bandage[1] + bandage[2]
                        conti_time = 0
                    else:
                        health = health + bandage[1]
                if health > max_health:
                    health = max_health
        if health <= 0:
            return -1
    return health